runtime/caml/domain.h

```diff
@@ -125,14 +125,14 @@ int caml_try_run_on_all_domains_with_spin_work(
   int sync,
   void (*handler)(caml_domain_state*, void*, int, caml_domain_state**),
   void* data,
-  void (*leader_setup)(caml_domain_state*),
+  void (*leader_setup)(caml_domain_state*, void *),
```

Comment:
Is the `void*` an argument to the `leader_setup` function? Document this?

Stepping back, `caml_try_run_on_all_domains_with_spin_work` type is quite complicated. It can do with some documentation.

Reply:
Yes, `data` is passed to both `handler` and `leader_setup`. Certainly more documentation would be good - there's a massive comment but it doesn't really explain the API.